main.py

```diff
@@ -33,47 +34,141 @@
 # create a Flask web app
 app = Flask(__name__)
 
-
-# handles the homepage for Hello Socks
+# decorator to use for routes that should only be accessed by authenticated users
+def auth_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        
+        # redirect to home if session is not present in request
+        auth_context = get_request_auth_context(request)
+        if auth_context is None:
+            print("Unauthenticated. Redirecting")
+            return redirect(url_for("index"))
+        
+        g.auth_context = auth_context
+        return f(*args, **kwargs)
+    return wrap
+
+# decorator for retrieving and passing along context on session user, if exists
+def with_auth_context(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        
+        g.auth_context = get_request_auth_context(request)
+        return f(*args, **kwargs)
+
+    return wrap
+
+
+# handles the homepage for Hello Socks depending on authentication state
 @app.route("/")
+@with_auth_context
 def index() -> str:
-    return render_template("loginOrSignUp.html")
+    if g.auth_context:
+        email = g.auth_context.get("user").emails[0].email
+        resp = make_response(render_template("loggedIn.html", email=email))
+        return resp
+    
+    resp = make_response(render_template("loginOrSignUp.html"))
+    resp.delete_cookie("stytch-session")
+    return resp
 
 
 # takes the email entered on the homepage and hits the stytch
 # loginOrCreateUser endpoint to send the user a magic link
 @app.route("/login_or_create_user", methods=["POST"])
 def login_or_create_user() -> str:
-    resp = stytch_client.magic_links.email.login_or_create(
-        email=request.form["email"],
-        login_magic_link_url=MAGIC_LINK_URL,
-        signup_magic_link_url=MAGIC_LINK_URL,
-    )
-
-    if resp.status_code != 200:
-        print(resp)
+
+    try:
+        resp = stytch_client.magic_links.email.login_or_create(
+            email=request.form["email"],
+            login_magic_link_url=MAGIC_LINK_URL,
+            signup_magic_link_url=MAGIC_LINK_URL,
+        )
+    except:
         return "something went wrong sending magic link"
-    return render_template("emailSent.html")
+ 
+    return render_template("emailSent.html", type="login")
 
 
 # This is the endpoint the link in the magic link hits.
 # It takes the token from the link's query params and hits the
 # stytch authenticate endpoint to verify the token is valid
 @app.route("/authenticate")
-def authenticate() -> str:
-    resp = stytch_client.magic_links.authenticate(request.args["token"])
+def authenticate():
 
-    if resp.status_code != 200:
-        print(resp)
+    try:
+        resp = stytch_client.magic_links.authenticate(token=request.args["token"], session_duration_minutes=5)
```

Comment:
I set the `session_duration_minutes` to 5 so you could easily test out what happens when the session expires -- can increase if that would potentially be annoying

Reply:
From experience, it is annoying. The default in example apps is typically 60 minutes for that reason!